torch/_dynamo/graph_region_tracker.py

```diff
@@ -0,0 +1,225 @@
+import functools
+import math
+from collections import defaultdict, deque
+from typing import Any, Callable, Deque, Dict, List, Optional, Set, Tuple
+
+import torch.fx
+from torch.utils._pytree import tree_flatten
+
+
+Node = torch.fx.Node
+Region = List[Node]
+IdenticalNodes = List[Node]
+GlobalStateKey = Tuple[bool, bool, int, bool, bool, torch.dtype, bool, bool, bool]
+NodeKey = Tuple[GlobalStateKey, str, int]
+
+
+def _get_tensor_props(node: Node) -> Optional[Tuple[torch.Size, torch.dtype]]:
+    if isinstance(node, torch.fx.Node):
+        value = node.meta.get("example_value", None)
+        if isinstance(value, torch.Tensor):
+            return value.shape, value.dtype
+    return None
+
+
+@functools.lru_cache(128)
+def _hash_node_inputs(node: Node) -> int:
+    flat_args, _ = tree_flatten(node.args)
+    sorted_kwargs = sorted(node.kwargs.items(), key=lambda x: x[0])
+    sorted_keys = tuple(sorted(node.kwargs.keys()))
+    flat_kwargs, _ = tree_flatten(sorted_kwargs)
+    all_args = flat_args + flat_kwargs
+    return hash((sorted_keys, tuple(_get_tensor_props(arg) for arg in all_args)))
+
+
+def get_global_state_key() -> GlobalStateKey:
```

Comment:
I wish there were some central location for all pytorch globals

Reply:
In particular, we're likely missing stuff in https://www.internalfb.com/code/fbsource/[5c15ca84a45e2002ddefb07f0e83228ccbc5c281]/fbcode/caffe2/aten/src/ATen/ThreadLocalState.cpp

Reply:
So I took a look at these, I think the only one that was missing and that dynamo supports putting in the graph is the saved_tensors_hooks TLS. I added this